python/preemo/worker/_artifact_manager.py

```diff
@@ -47,22 +57,68 @@ def get_artifacts(self, artifact_ids: List[ArtifactId]) -> List[bytes]:
 
 
 class ArtifactManager:
-    def __init__(self, *, messaging_client: IMessagingClient) -> None:
+    @staticmethod
+    def _calculate_part_count(*, content_length: int, part_size_threshold: int) -> int:
+        return math.ceil(content_length / part_size_threshold)
+
+    @staticmethod
+    def _upload_content(*, content: memoryview, url: str) -> None:
+        # TODO(adrian@preemo.io, 04/06/2023): how to force gzip
+        response = requests.put(url=url, data=content)
+
+        # TODO(adrian@preemo.io, 04/06/2023): should probably retry
+        if not response.ok:
+            raise Exception(f"unexpected response while uploading: {response}")
+
+    @staticmethod
+    def _download_content(*, url: str) -> bytes:
+        # TODO(adrian@preemo.io, 04/06/2023): how to force (un)gzip
```

Comment:
according to ChatGPT, we'll be able to do something like this when we create the signed urls:
```
signed_url = blob.generate_signed_url(
    expiration=datetime.timedelta(minutes=10),
    method='GET',
    content_type='application/pdf',
    response_type='attachment; filename=myfile.pdf',
    response_encoding='gzip',
    cache_control='public, max-age=3600',
    version='v4'
)
```
When I ask about the response encoding it says:

> The Response Content Encoding option for a signed URL determines the encoding of the content that will be served when the URL is used.
> 
> Content encoding is a technique for compressing the content of an HTTP response to reduce the amount of data that needs to be transferred over the network. This can help to reduce the response time for the request and save bandwidth.
> 
> When you specify a value for the response_encoding parameter when creating a signed URL, the Content-Encoding header is set for the response to the HTTP request. This informs the client that the content is compressed and needs to be decompressed before it can be used.
> 
> For example, if you set the response_encoding to gzip, the Content-Encoding header of the HTTP response will be set to gzip. The client will then automatically decompress the content using the gzip algorithm before processing it.
> 
> The available options for response_encoding include:
> 
> identity: This indicates that the content is not encoded or compressed.
> 
> gzip: This indicates that the content is compressed using the gzip algorithm.
> 
> deflate: This indicates that the content is compressed using the deflate algorithm.
> 
> br: This indicates that the content is compressed using the Brotli algorithm.
> 
> Note that the client must support the encoding method used in the response_encoding parameter in order to decompress the content.

So it might just work? It probably makes more sense for me to just set that header explicitly here